compiler-core/src/language_server/engine.rs

```diff
@@ -68,6 +68,9 @@ pub struct LanguageServerEngine<IO, Reporter> {
     // the usual request-response loop.
     progress_reporter: Reporter,
 
+    // Keep track of unused value from warnings
+    unused_values: Vec<SrcSpan>,
```

Comment:
Rather than storing unused values locations in the engine the engine can on-demand traverse the warnings in the module interface to find the locations.

Reply:
Thank you for the suggestion! It looks way cleaner now!